### DISCUSSION TITLE
Allow icon packs to switch colors between the system's light & dark mode

### DISCUSSION THREAD

**Discussion by Donnnno** [breezy-weather/breezy-weather#1224] [Q&A] [answered]
### Steps to reproduce

Hi, we're currently working on implementing an icon pack for breezy into [Arcticons](https://github.com/Arcticons-Team/Arcticons/pull/2201). (loving how easy it is to implement)

However, in my testings, I noticed that the icon colors won't change despite it having a color option that should respect the system theme: `android:strokeColor="@color/icon_color"
`

That results in some unfortunate white spaces when the widget and apps are set to light mode (in case the user wants that)

![image](https://github.com/user-attachments/assets/3c9e86a5-f5d1-4877-a4be-8a76a410b706)
![image](https://github.com/user-attachments/assets/392d1c5e-8677-4361-ae9d-6f9832be7957)

Would it be possible to make the icons respect the system's (or app's) light & dark mode?

### Expected behavior

The icons should change color with the system theme

### Actual behavior

Nothing happens

### Weather source used

Default

### Breezy Weather version

5.2.5_freenet & 5.2.6_freenet

### Android version

Android 13

### Device

Pixel 7a emulator

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue.
- [X] I checked that my issue is not covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document.
- [X] I have updated the app to **[version 5.2.6 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)** and I can reproduce the issue in this version.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language. I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment, marked as answer]
Works fine on my side with the default icon pack. Did you have a look at how it is designed?
If you go to the details page of the default icon pack, you will see that the color will automatically switch whether you use a light or dark theme.

I never designed an icon pack, but I can see a few things that you may miss:
- The XML contains `android:fillColor="#000"`; I seem to remember we tell Android to change this `fillColor`
- There are three PNG icons: `mini_dark`, `mini_grey` and `mini_light`

**Reply by Donnnno** [reply]
Thank you for your quick reply! 
It was really stupid of me, but it turned out I switched the light & dark values!

![image](https://github.com/user-attachments/assets/1d7531f1-8f2d-4adf-81a5-59129fcd5f3d)
![image](https://github.com/user-attachments/assets/cb5d9ff3-3eca-4087-8a31-c55c97fd6e87)

**Comment by someone-aka-sum1** [comment]
With Arcticons Black, icons in the persistent notification were white (on light, with persistent light theme) for me. I'm not sure why and I didn't notice this problem with other apps (then again, they might not be using icons). As a workaround, switching to Arcticons Day & Night made them black.
